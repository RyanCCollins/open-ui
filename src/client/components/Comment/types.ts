import * as React from 'react';
import Comment from '../';

export interface CommentProps extends React.Props<typeof Comment> {
  author?: string;
  body?: string;
  picUrl?: string;
}
