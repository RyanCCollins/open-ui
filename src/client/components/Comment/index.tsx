import * as React from 'react';
import { CommentProps } from './types';
import {
  Wrapper,
  CommentWrapper,
  PicWrapper,
  Pic,
  CommentText,
  Footer,
} from './styles';

export default function Comment({
  author,
  body,
  picUrl,
}: CommentProps): JSX.Element {
  const url = picUrl || 'https://github.com/RyanCCollins/cdn/blob/master/misc/missing.png?raw=true';
  return (
    <Wrapper>
      <PicWrapper>
        <Pic url={url} />
      </PicWrapper>
      <CommentWrapper>
        <CommentText>{body}</CommentText>
        <Footer>
          {author} | Aug 24, 2014 @ 2:35 PM
        </Footer>
      </CommentWrapper>
    </Wrapper>
  );
}
