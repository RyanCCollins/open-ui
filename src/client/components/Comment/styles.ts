const styled = require('styled-components').default;

export const Wrapper = styled.div`
  min-height: 5.3125rem;
  margin-bottom: 1.25rem;
  width: 100%;
  display: flex;
  flex-direction: row;
`;

export const CommentWrapper = styled.div`
  padding: 1rem;
  background-color: #fff;
  vertical-align: top;
  border-radius: 0.1875rem;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.08);
  flex: 1;
`;

export const CommentText = styled.p`
  color: #555f77;
  margin-bottom: 1.25rem;
  line-height: 1.3125rem;
`;

export const PicWrapper = styled.div`
  padding-top: 0.625rem;
  width: 3.5rem;
`;

export const Pic = styled.div`
  height: 2.25rem;
  width: 2.25rem;
  border-radius: 50%;
  background-size: contain;
  background-image: url('${(props) => props.url}');
`;

export const Footer = styled.footer`
  color: #acb4c2;
  font-size: 0.875rem;
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  flex-direction: row;
`;
