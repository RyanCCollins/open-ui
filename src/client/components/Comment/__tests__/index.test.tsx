import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Comment from '../';
import { commentProps, commentPicProps } from './__mocks__/commentMocks.mock';

describe('<Comment /> default', () => {
  it('should render a plain comment with required props', () => {
    const wrapper = shallow(
      <Comment {...commentProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a plain comment with other props', () => {
    const wrapper = shallow(
      <Comment {...commentPicProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
