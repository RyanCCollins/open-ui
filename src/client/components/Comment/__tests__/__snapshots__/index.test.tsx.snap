// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> default should render a plain comment with other props 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      url="https://github.com/RyanCCollins/cdn/blob/master/misc/missing.png?raw=true"
    />
  </styled.div>
  <styled.div>
    <styled.p>
      Comment Body
    </styled.p>
    <styled.footer>
      Comment Author
       | Aug 24, 2014 @ 2:35 PM
    </styled.footer>
  </styled.div>
</styled.div>
`;

exports[`<Comment /> default should render a plain comment with required props 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      url="https://github.com/RyanCCollins/cdn/blob/master/misc/missing.png?raw=true"
    />
  </styled.div>
  <styled.div>
    <styled.p>
      Comment Body
    </styled.p>
    <styled.footer>
      Comment Author
       | Aug 24, 2014 @ 2:35 PM
    </styled.footer>
  </styled.div>
</styled.div>
`;
